### PR TITLE
Upgrade kata-containers to version 3.1.0

### PR DESCRIPTION
**Pull request by Redent0r**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Upgrades kata-containers to 3.1.0 and fix patches

###### Change Log  

- kata-containers

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

1. Buddy build package: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=349161&view=results
2. Installed on running cluster
3. Re-ran kata-osbuilder.sh due to https://microsoft.visualstudio.com/OS/_workitems/edit/43883105 (This won't be necessary and will work out of the box when built through the pipelines)
4. Passed 6/6 container tests. Also able to run ubuntu using kata runtime
[container_test_results.txt](https://github.com/microsoft/CBL-Mariner/files/11336339/runner.txt)
-----------------
1. Created test image with new kata-containers: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=350928&view=results
2. Ran through conformance+container pipeline and got good results: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=350928&view=results
3. Perf run: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=351026&view=results
Values seems to be generally on par with previous perf runs
image

### CODE REVIEW

File: SPECS/kata-containers/runtime-Re-generate-the-client-code.patch [hunk 1]
```diff
@@ -1,1401 +0,0 @@
-From 2edfd719d46f26c85882944c64ebed3f315d72df Mon Sep 17 00:00:00 2001
```
Comment (Redent0r) [inline SPECS/kata-containers/runtime-Re-generate-the-client-code.patch:1]: [This](https://github.com/kata-containers/kata-containers/commit/2edfd719d46f26c85882944c64ebed3f315d72df) is cherry picked from https://github.com/kata-containers/kata-containers/commit/5d63fcf344808495d3a67f1f718a7f0956bac87f, which is already merged in 3.1.0
image

File: SPECS/kata-containers/runtime-Support-for-AMD-SEV-SNP-VMs.patch [hunk 2]
```diff
@@ -1,268 +0,0 @@
-From 22bda0838c77853ed92546308b59b6500083bb4e Mon Sep 17 00:00:00 2001
```
Comment (Redent0r) [inline SPECS/kata-containers/runtime-Support-for-AMD-SEV-SNP-VMs.patch:1]: https://github.com/kata-containers/kata-containers/commit/22bda0838c77853ed92546308b59b6500083bb4e merged in 3.1.0 
image

File: SPECS/kata-containers/runtime-clh-Use-the-new-API-to-boot-with-TDX-firmware-td-shim.patch [hunk 3]
```diff
@@ -1,35 +0,0 @@
-From 067e2b1e33b34b6bd0a55246f8f3db36dce3b3f2 Mon Sep 17 00:00:00 2001
```
Comment (Redent0r) [inline SPECS/kata-containers/runtime-clh-Use-the-new-API-to-boot-with-TDX-firmware-td-shim.patch:1]: https://github.com/kata-containers/kata-containers/commit/067e2b1e33b34b6bd0a55246f8f3db36dce3b3f2 merged in 3.1.0 
image

File: SPECS/kata-containers/versions-Update-Cloud-Hypervisor.patch [hunk 4]
```diff
@@ -1,975 +0,0 @@
-From 9d286af7b4543b0af9e6db53c6c84e0731f61c09 Mon Sep 17 00:00:00 2001
```
Comment (Redent0r) [inline SPECS/kata-containers/versions-Update-Cloud-Hypervisor.patch:1]: https://github.com/kata-containers/kata-containers/commit/9d286af7b4543b0af9e6db53c6c84e0731f61c09 merged in 3.1.0
image

**Review by dallasd1 — APPROVED** [review]

**Review by sprt — APPROVED** [review]
Perf results LGTM!

**Pull request merged**